src/Policies.cpp

```diff
@@ -336,7 +357,16 @@ void PolicyManager::remove_policy(const std::string &name)
         _registry->handler_manager()->module_remove(name);
     }
 
-    if (!policy->input_stream()->policies_count()) {
+    if (policy->input_stream()->policies_count() == 1) {
+        auto input_resources_name = input_name + "-resources";
+        if (_map.count(input_resources_name) != 0) {
+            auto resources_policy = _map[input_resources_name].get();
+            resources_policy->stop();
+            _registry->handler_manager()->module_remove(input_resources_name);
+            _registry->input_manager()->module_remove(input_name);
+            _map.erase(input_resources_name);
+        }
+    } else if (!policy->input_stream()->policies_count()) {
```

Comment:
this shouldn't be "else if", just "if" right, because if we just removed the input resources policy, this will be 0 and we want to remove the input

Reply:
I mean I was removing in both == 1 and == 0, but changed the logic now